nuxeo-features/nuxeo-platform-imaging/nuxeo-platform-imaging-jsf/src/test/java/org/nuxeo/ecm/imaging/recompute/lang/test/PerformingImageActionTest.java

```diff
@@ -0,0 +1,149 @@
+
+    public static final String USER_WITH_READ_WRITE_PERMISSION = "james";
+
+    public static final String USER_WITH_EVERYTHING_PERMISSION = "anotherAdmin";
+
+    public static final String ROTATE_PICTURE_FILTER_NAME = "canRotatePicture";
+
+    @Inject
+    protected ActionManager actionService;
+
+    @Inject
+    protected CoreSession coreSession;
+
+    @Inject
+    protected CoreFeature coreFeature;
+
+    protected MockFacesContext facesContext;
+
+    @Before
+    public void initRepository() {
+        facesContext = new MockFacesContext();
+        facesContext.setCurrent();
+
+        DocumentModel document = coreSession.createDocumentModel("/", PICTURE_NAME, PICTURE_DOC_TYPE);
+        coreSession.createDocument(document);
+
+        ACE[] aces = { getACE(USER_WITH_READ_PERMISSION, SecurityConstants.READ),
+                getACE(USER_WITH_READ_WRITE_PERMISSION, SecurityConstants.READ_WRITE),
+                getACE(USER_WITH_EVERYTHING_PERMISSION, SecurityConstants.EVERYTHING) };
+
+        ACLImpl acl = new ACLImpl();
+        acl.setACEs(aces);
+
+        ACPImpl acp = new ACPImpl();
+        acp.addACL(acl);
+        document.setACP(acp, Boolean.FALSE);
+
+        coreSession.save();
+    }
+
+    @Test
+    public void shouldBeRotatableByTheCreator() {
+        boolean canRotatePicture = actionService.checkFilter(ROTATE_PICTURE_FILTER_NAME, getActionContext(coreSession));
+        assertTrue(canRotatePicture);
+    }
+
+    @Test
+    public void shouldNotBeRotatableByUnauthorizedUser() {
+        try (CloseableCoreSession session = coreFeature.openCoreSession(USER_WITH_READ_PERMISSION)) {
+            boolean canRotatePicture = actionService.checkFilter(ROTATE_PICTURE_FILTER_NAME, getActionContext(session));
+            assertFalse(canRotatePicture);
+        }
+    }
+
+    @Test
+    public void shouldBeRotatableByAuthorizedUser() {
+        try (CloseableCoreSession session = coreFeature.openCoreSession(USER_WITH_READ_WRITE_PERMISSION)) {
+            boolean canRotatePicture = actionService.checkFilter(ROTATE_PICTURE_FILTER_NAME, getActionContext(session));
+            assertTrue(canRotatePicture);
+        }
+    }
+
+    @Test
+    public void shouldBeRotatableBySuperUser() {
+        try (CloseableCoreSession session = coreFeature.openCoreSession(USER_WITH_EVERYTHING_PERMISSION)) {
+            boolean canRotatePicture = actionService.checkFilter(ROTATE_PICTURE_FILTER_NAME, getActionContext(session));
+            assertTrue(canRotatePicture);
+        }
+    }
+
+    protected ActionContext getActionContext(CoreSession session) {
+        ActionContext context = new JSFActionContext(facesContext);
+        context.setCurrentPrincipal(session.getPrincipal());
+        context.setDocumentManager(session);
+
+        DocumentModelList documents = session.query(
+                String.format("Select * From %s where ecm:name='%s'", PICTURE_DOC_TYPE, PICTURE_NAME));
+        context.setCurrentDocument(documents.get(0));
```

Comment:
Why not just
```
DocumentModel document = session.getDocument(new PathRef("/" + PICTURE_NAME));
```
?

Reply:
You could even just store the document ref as a field variable in the `initRepository`, and use it with `CoreSession#getDocument`.